core/arch/aarch64/proc.c

```diff
@@ -122,14 +122,20 @@ get_processor_specific_info(void)
         /* This RDVL instruction is inserted as raw hex because we don't build
          * with SVE enabled: i.e. not -march=armv8-a+sve, so that we can run a
          * single DynamoRIO release on both SVE and non-SVE h/w.
+         * i#6852: Some compiler toolchains were observed to generate incorrect
+         * asm where the following was no longer gated by the above if-condition.
+         * This may perhaps be a compiler implementation specific bug. To
```

Comment:
I'm not sure this would be considered a bug: compilers are free to move things around so long as the observable effects are the same and it could be that SIGILL is not on the list of things the compiler has to consider. It can hoist things out of loops and `if` blocks if it thinks it doesn't affect things.

Reply:
Done. Reworded to not blame the compiler.